src/python/pants/backend/visibility/rule_types_test.py

```diff
@@ -787,20 +787,14 @@ def test_file_specific_rules(rule_runner: RuleRunner) -> None:
             "src/lib/pub/BUILD": dedent(
                 """
                 __dependents_rules__(
-                  # Allow all to depend on files from lib/pub/
-                  # TODO: support to except one file. work around now is to have the rule in
-                  # src/app/BUILD
+                  # Allow all to depend on files from lib/pub/ except for one file in particular.
+                  ("[/exception.txt]", "//src/lib/**", "!*"),
```

Comment:
The selector (`"[/exception.txt]"`) is evaluated in the context of the target from the current BUILD file, where as the rule glob (`"//src/lib/**"`) is evaluated against the dependency from another BUILD file (the other end of the dependency link).